File: IM.eSuite.Web/src/app/modules/administration/models/applicationUserGroup.ts
export class ApplicationUserGroup {
    id: number;
    name: string;
    applicationId: number;
    
    constructor(applicationUserGroup?:ApplicationUserGroup){
        if(!applicationUserGroup){
            return;
        }

        this.id = applicationUserGroup.id;
        this.name = applicationUserGroup.name;
        this.applicationId = applicationUserGroup.applicationId;
    }
}

File: IM.eSuite.Domain/Model/ApplicationUserGroup.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using IM.NETCore.Domain;

namespace IM.eSuite.Domain
{
    public class ApplicationUserGroup: EntityBase
    {
        [Required]
        [MaxLength(50)]
        public string Name { get; set; }        

        public int ApplicationId { get; set; }
        
        // Navigation properties        
        public virtual ICollection<UserGroupUser> Users { get; set; }
        public virtual ICollection<ApplicationUserGroupRole> UserGroupRoles { get; set; }
        public virtual Application Application { get; set; }
    }
}